sdk/core/System.ClientModel/src/Convenience/AsyncCollectionResultOfT.cs

```diff
@@ -4,41 +4,38 @@
 using System.ClientModel.Primitives;
 using System.Collections.Generic;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace System.ClientModel;
 
+#pragma warning disable CS1591 // public XML comments
 /// <summary>
 /// Represents a collection of values returned from a cloud service operation.
 /// The collection values may be returned by one or more service responses.
 /// </summary>
-public abstract class AsyncCollectionResult<T> : ClientResult, IAsyncEnumerable<T>
+public abstract class AsyncCollectionResult<T> : AsyncCollectionResult, IAsyncEnumerable<T>
 {
-    /// <summary>
-    /// Create a new instance of <see cref="AsyncCollectionResult{T}"/>.
-    /// </summary>
-    /// <remarks>If no <see cref="PipelineResponse"/> is provided when the
-    /// <see cref="ClientResult"/> instance is created, it is expected that
-    /// a derived type will call <see cref="ClientResult.SetRawResponse(PipelineResponse)"/>
-    /// prior to a user calling <see cref="ClientResult.GetRawResponse"/>.
-    /// This constructor is indended for use by collection implementations that
-    /// postpone sending a request until <see cref="GetAsyncEnumerator(CancellationToken)"/>
-    /// is called. Such implementations will typically be returned from client
-    /// convenience methods so that callers of the methods don't need to
-    /// dispose the return value. </remarks>
-    protected internal AsyncCollectionResult() : base()
+    protected internal AsyncCollectionResult()
     {
     }
 
-    /// <summary>
-    /// Create a new instance of <see cref="AsyncCollectionResult{T}"/>.
-    /// </summary>
-    /// <param name="response">The <see cref="PipelineResponse"/> holding the
-    /// items in the collection, or the first set of the items in the collection.
-    /// </param>
-    protected internal AsyncCollectionResult(PipelineResponse response) : base(response)
+    /// <inheritdoc/>
+    public async IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken = default)
     {
+        await foreach (ClientResult page in GetRawPagesAsync().ConfigureAwait(false).WithCancellation(cancellationToken))
+        {
+            await foreach (T value in GetValuesFromPageAsync(page).ConfigureAwait(false).WithCancellation(cancellationToken))
+            {
+                yield return value;
+            }
+        }
     }
 
-    /// <inheritdoc/>
-    public abstract IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken = default);
+    /// <summary>
+    /// Get a collection of the values returned in a page response.
+    /// </summary>
```

Comment:
nit:  Won't trigger a failure with the warning restored, but we should fill these in so that they're meaningful.

Reply:
Yep, I was planning to loop back over anything I'd disabled CS1591 and write those fresh.  Thanks!